### PR TITLE
Add config updater

### PR DESCRIPTION
**Pull request by eyurtsev**
Add config updater

### CODE REVIEW

File: tests/unit_tests/test_server_client.py [hunk 1]
```diff
@@ -405,6 +405,44 @@ async def mul_2(x: int) -> int:
...
+        if "metadata" in config:
+            config["metadata"] = config["metadata"].copy()
+        else:
+            config["metadata"] = {}
+        config["metadata"]["headers"] = request.headers
+        return config
+
+    server_runnable = RunnableLambda(add_one)
+
+    add_routes(app, server_runnable, path="/add_one", config_updater=config_updater)
```
Comment (jacoblee93) [inline tests/unit_tests/test_server_client.py:432]: Maybe call this `config_extractor`?

And how do I actually call this as an end user? A `config` body param?

Comment (eyurtsev) [reply tests/unit_tests/test_server_client.py:432]: This is for picking up information from request headers / cookies if needed for some reason and placing it inside the config.

**Comment by eyurtsev** [comment]
We're holding off for a bit until we see a real use case

**Comment by eyurtsev** [comment]
Handled by https://github.com/langchain-ai/langserve/pull/161

**Pull request closed without merging**